components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -478,6 +489,34 @@ && isStepHasMultiOption(context)
         }
     }
 
+    private boolean checkIfApplicationEnabled(HttpServletRequest request, AuthenticationContext context)
+            throws FrameworkException {
+
+        String type = request.getParameter(TYPE);
+        String clientId = request.getParameter(CLIENT_ID);
+        if (StringUtils.isBlank(type)) {
+            type = context.getRequestType();
+        }
+        if (StringUtils.isBlank(clientId)) {
+            clientId = context.getRelyingParty();
+        }
+        ServiceProvider serviceProvider = getServiceProvider(type, clientId, getTenantDomain(request));
+        if (serviceProvider == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Service provider is null for client id: " + clientId);
```

Comment:
client id implies the oauth apps
Is that saml app case also handled here?

Reply:
Yes. Both apps will be handled here.